R/internal_utility.R

```diff
@@ -105,14 +106,34 @@
     } 
     
     if (length(res) == 1) {
-      rep(res, n)
+      rep(res, times = .n)
     } else {
       res
     }
   }
-
+  
+  ### Get functions from calling environment - added 20240724
+  
+  if (length(ls(envir)) != 0) {  
+    
+    all_objects <- ls(envir)
+    
+    function_objects <- all_objects[
+      sapply(
+        all_objects,
+        function(obj) {
+          is.function(get(obj, envir = envir))
+        })
+    ]
+    
+    functions_list <- mget(function_objects, envir = envir)
+    list2env(functions_list, envir = environment())
```

Comment:
I think you should add these to the same env used for the dot vars:
```suggestion
    list2env(functions_list, envir = e)
```

Reply:
I've made the change - that seems even safer. I will merge (and then submit to CRAN) once it passes our tests.